annotation-processing/src/main/java/com/exactpro/epfast/annotation/processing/FastProcessor.java

```diff
@@ -9,122 +12,253 @@
             }
             return true;
         }
-
         return false;
     }
 
-    private HashMap<String, Element> collectFastTypes(RoundEnvironment roundEnv) {
-        HashMap<String, ArrayList<Element>> nameClassMap = new HashMap<>();
+    private void removeAndReportIfRootEqualsFastPackage(FastCreator fastCreator) {
+        AbstractFastPackage rootPackage = fastCreator.getRootPackage();
+        if (rootPackage == null) {
+            return;
+        }
+        Collection<AbstractFastPackage> fastPackages = fastCreator.getAllFastPackages();
+        long duplicates = fastPackages.stream().filter(
+            fastPackage -> fastPackage.getName().equals(rootPackage.getName()) && !fastPackage.isRootPackage()).count();
+        if (duplicates > 0) {
+            fastCreator.removePackage(rootPackage.getName());
+            errorReporter.reportRootEqualsFastPackage(rootPackage.getName());
+        }
+    }
+
+    private boolean containsDuplicateFastTypes(AbstractFastPackage fastPackage) {
+        List<Element> fastTypes = fastPackage.getFastTypes();
+        HashMap<String, ArrayList<Element>> fastTypeNames = collectFastTypeNames(fastTypes);
+        AtomicBoolean invalid = new AtomicBoolean(false);
+        fastTypeNames.entrySet().stream().filter(entry -> entry.getValue().size() > 1)
+            .forEach(entry -> {
+                String fastTypeName = entry.getKey();
+                ArrayList<Element> elements = entry.getValue();
+                errorReporter.reportDuplicateFastTypes(fastTypeName, elements);
+                invalid.set(true);
+            });
+        return invalid.get();
+    }
+
+    private HashMap<String, ArrayList<Element>> collectFastTypeNames(List<Element> fastTypes) {
+        HashMap<String, ArrayList<Element>> fastTypeMap = new HashMap<>();
+        fastTypes.forEach(element -> {
+            String fastTypeName = getFastTypeName(element);
+            fastTypeMap.computeIfAbsent(fastTypeName, key -> new ArrayList<>()).add(element);
+        });
+        return fastTypeMap;
+    }
+
+    private void removeAndReportInvalidPackages(HashMap<String, HashSet<Element>> allPackages,
+                                                FastCreator fastCreator) {
+        Iterator<Map.Entry<String, HashSet<Element>>> iterator = allPackages.entrySet().iterator();
+        while (iterator.hasNext()) {
+            Map.Entry<String, HashSet<Element>> entry = iterator.next();
+            String fastPackage = entry.getKey();
+            HashSet<Element> elements = entry.getValue();
+            if (!isValidPackage(fastPackage)) {
+                elements.forEach(element -> errorReporter.reportInvalidPackage(fastPackage, element));
+                fastCreator.removePackage(fastPackage);
+                iterator.remove();
+            }
+        }
+    }
+
+    private void removeAndReportDuplicates(HashMap<String, HashSet<Element>> allPackages, FastCreator fastCreator) {
```

Comment:
Don't remove anything. Add method to error reporter which checks if any errors were reported. If it's set, don't generate any files.